Make banner detection non-greedy in ios_banner

kalimsshar opened the pull request:
##### SUMMARY
<!--- Describe the change below, including rationale and design decisions -->
Bug #63091 is caused because regex used in ios_banner is greedy.
If multiple banners are present in the configuration, a "show running-config | begin banner exec" will return the end of the configuration, including other configured banners.

<!--- HINT: Include "Fixes #nnn" if you are fixing an existing issue -->
Fixes #63091

##### ISSUE TYPE
<!--- Pick one below and delete the rest -->
- Bugfix Pull Request

##### COMPONENT NAME
<!--- Write the short name of the module, plugin, task or feature below -->
ios_banner

##### ADDITIONAL INFORMATION
<!--- Include additional information to help people understand the change here -->
Only change is to make the regex that search for a banner non greedy (.*?)

<!--- A step-by-step reproduction of the problem is helpful if there is no related issue -->
See bug #63091 for reproducing the bug.

<!--- Paste verbatim command output below, e.g. before and after your change -->
```paste below

```

kalimsshar committed:
Make banner detection non-greedy in ios_banner

GomathiselviS commented:
@kalimsshar Thank you for creating PR and working on it. The fix looks fine. Can you please add testcase to cover this scenario ?

GomathiselviS commented:
needs_info

kalimsshar commented:
@GomathiselviS I'm working on it, but I'm having a few difficulties to install a working test environment. I'll update soon.

kalimsshar committed:
Added ios_banner unit test to detect bug #63091

GomathiselviS commented:
@kalimsshar Thank you for adding the unit testcase. I shall review the same and get back to you if needed. Meanwhile, can you please add the relevant  integration testcase to test/integration/targets/ios_banner/tests/cli ? These testcases must be straight forward. Please let me know if you need any assistance.

pabelanger commented:
This failure here, looks to be an issue with our Fedora-29 images. Already have a fix in progress.

pabelanger commented:
recheck

kalimsshar committed:
Corrected PEP8 errors

kalimsshar committed:
Added integration test

kalimsshar commented:
@GomathiselviS Sorry for the misunderstanding, I read the integration tests documentation and I think I added the correct scenario. I also corrected the PEP8 errors in my unit testcase.

kalimsshar committed:
Corrected typo in integration test

GomathiselviS commented:
recheck

kalimsshar commented:
Hey @GomathiselviS. What is the problem with the Shippable check? It seems to me that it is a problem on the check platform and it was not re-checked since the 1st fail. Is there anything I can fix on my end?
FYI, I'm currently mostly AFK, will be more available in November.

GomathiselviS commented:
recheck

GomathiselviS commented:
@kalimsshar The tests passed after closing and reopening the PR. I have to  check , why the failure happened in the first place.
